shell/src/stats/memory.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use nix;
+use page_size;
+use serde::Serialize;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use std::error;
+use std::fmt;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::process::Command;
+use core::str::{from_utf8};
+
+pub type MemoryStatsResult<T> = std::result::Result<T, MemoryStatsError>;
+
+#[derive(Serialize, PartialEq, Clone, Debug)]
+pub struct LinuxData {
+    page_size: usize,
+    size: String,
+    resident: String,
+    shared: String,
+    text: String,
+    lib: String,
+    data: String,
+    dt: String
+}
+
+#[derive(Serialize, PartialEq, Clone, Debug)]
+pub struct DarwinOsData {
```

Comment:
I noticed that `LinuxData` and `DarwinOsData` structures have different shape. This will be a problem for the consumers of the rpc api. Rpc api should be consistent and should not change between platforms eg. between linux and mac.

Reply:
check spec at http://tezos.gitlab.io/api/rpc.html#get-stats-memory

there is different output for mac and linux, if we want be compatible with them

Reply:
cool, didn't know that